controllers/argocd/statefulset.go

```diff
@@ -591,3 +598,31 @@ func updateNodePlacementStateful(existing *appsv1.StatefulSet, ss *appsv1.Statef
 		*changed = true
 	}
 }
+
+//to fetch the broken pod of StatefulSet if any, this function intends to delete the broken pod that cannot
+//be restarted due to known kubernetes issue https://github.com/kubernetes/kubernetes/issues/67250
+func containsBrokenPod(ss *appsv1.StatefulSet, cr *argoprojv1a1.ArgoCD) bool {
+
+	k8s, err := initK8sClient()
+	if err != nil {
+		log.Error(err, "Failed to initialize k8s client")
+	}
+	brokenPod := false
+	labelSel := common.ArgoCDKeyName + " = " + fmt.Sprintf("%s-%s", cr.Name, "application-controller")
+	po, err := k8s.CoreV1().Pods(ss.Namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: labelSel})
+	if err != nil {
+		log.Error(err, "Failed to list Pods")
+	}
+
+	if len(po.Items) > 0 {
+		for _, p := range po.Items {
```

Comment:
Do you think we need to iterate through the list of pods and break on the first pod in ImagePullBackOff state ? Can we instead verify this condition on the first pod as the behavior should be same for all the pods.

Reply:
Ok, in that case I'll remove the for loop and break, but we will still need the two if conditions